super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,31 +231,65 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
-				if( c == quoteChar ) {
+
+				if( c == quoteEscapeChar ) {
 					int nextCharIndex = charIndex + 1;
 					boolean availableCharacters = nextCharIndex < line.length();
 					boolean nextCharIsQuote = availableCharacters && line.charAt(nextCharIndex) == quoteChar;
+					boolean nextCharIsEscapeQuoteChar = availableCharacters && line.charAt(nextCharIndex) == quoteEscapeChar;
+
 					if( nextCharIsQuote ) {
 						/*
-						 * An escaped quote (""). Add a single quote, then move the cursor so the next iteration of the
-						 * loop will read the character following the escaped quote.
+						 * An escaped quote (e.g. "" or \"). Skip over the escape char, and add
+						 * the following quote char as part of the column;
+						 */
+						charIndex++;
+						currentColumn.append(line.charAt(nextCharIndex));
+					} else if( nextCharIsEscapeQuoteChar ) {
+						/*
+						 * A double escape (normally \\). Save the escape char, then continue to
+						 * next character.
 						 */
 						currentColumn.append(c);
 						charIndex++;
-						
-					} else {
+					} else if( quoteEscapeChar == quoteChar ) {
```

Comment:
Yes. Indeed. And now i know why I dont like this code anymore.  We are doing too many things at the same time. We are drowning in details.  How about we introduce 2 methods. `lexOutsideQuote` and lexInsideQuote. Each return a tokengroup (character, quote,eol,oef), the tokenized value and the new index. As input it takes the index and the current line.

Thus we separate the logic for tokenizing and state transition.

We have to be clever as not to loose toomuch to performance. There are performance test to be run before and after this change. The returned instance can be reused so each call to the lex does not instantiate. Or perhaps the index and line should be inserted into a state obejct such that the lax can mutate the values directly. Eg charIndex